protos/flyteidl/event/event.proto

```diff
@@ -62,8 +62,20 @@ message NodeExecutionEvent {
         WorkflowNodeMetadata workflow_node_metadata = 8;
     }
 
-    // Specifies which task (if any) launched this node.
+    // [To be deprecated] Specifies which task (if any) launched this node.
     ParentTaskExecutionMetadata parent_task_metadata         = 9;
+
+    // Specifies the parent node of the current node execution. Node executions at level zero will not have a parent node.
+    ParentNodeExecutionMetadata parent_node_metadata	= 10;
+
+    // Group identifier to indicate arbitrary grouping like retries, sub workflow
+    string group_id = 11;
+
+    // Identifier of the node in the original workflow/graph
+    string node_id = 12;
```

Comment:
isn't this originally captured by the node_id in the identifier?

Reply:
Yes.  That will now become unique identifier. This will point to the original node id in the graph.

Reply:
I didn't see this field exposed in any of the NodeExecution structures above. Is that intentional? As it is, I don't see a way for consumers of NodeExecution to know which node in the graph it relates to.